### PR TITLE
Fill out "Sourcing from the filesystem" stub article

### PR DESCRIPTION
**Pull request by caseychien**
This issue (#8730) is part of the content migration project [#8103](#8103), which is meant to take existing content and repurpose it so it's easier to find in the docs.

### CODE REVIEW

File: docs/docs/sourcing-from-the-filesystem.md [hunk 1]
```diff
@@ -2,7 +2,105 @@
...
-This is a stub. Help our community expand it.
+This guide will walk you through sourcing from the filesystem.
 
-Please use the [Gatsby Style Guide](/docs/gatsby-style-guide/) to ensure your
-pull request gets accepted.
+## Setup
+
+This guide assumes that you have a Gatsby project set up. If you need to set up a project, please reference the [Quick Start Guide](https://github.com/gatsbyjs/gatsby/tree/master/docs). In addition, it will be useful if you are familiar with Graph_i_QL, a tool that helps you structure your queries correctly.
+
+## Introducing Graph_i_QL
```
Comment (amberleyromo) [inline docs/docs/sourcing-from-the-filesystem.md]: I don't really think this section is necessary for this doc. I'll explain further in a different comment in this review.

File: docs/docs/sourcing-from-the-filesystem.md [hunk 2]
```diff
@@ -2,7 +2,105 @@
...
+
+You can access it when your site's development server is running—normally at
+<http://localhost:8000/___graphql>.
+
+<video controls="controls" autoplay="true" loop="true">
+  <source type="video/mp4" src="/graphiql-explore.mp4"></source>
+  <p>Your browser does not support the video element.</p>
+</video>
+
+Here you poke around the built-in `Site` "type" and see what fields are available
```
Comment (amberleyromo) [inline docs/docs/sourcing-from-the-filesystem.md]: Same thing, this makes no sense out of context of the tutorial.

File: docs/docs/sourcing-from-the-filesystem.md [hunk 3]
```diff
@@ -2,7 +2,105 @@
...
+the autocomplete window and <kbd>Ctrl + Enter</kbd> to run the GraphQL query. You'll be
+using Graph_i_QL a lot more through the remainder of the tutorial.
+
+## Sourcing plugins
+
+Data in Gatsby sites can come from anywhere: APIs, databases, CMSs,
+local files, etc.
+
+Source plugins fetch data from their source. E.g. the filesystem source plugin
+knows how to fetch data from the file system. The WordPress plugin knows how to
```
Comment (amberleyromo) [inline docs/docs/sourcing-from-the-filesystem.md]: I think delete the last two sentences of this paragraph -- no need.

File: docs/docs/sourcing-from-the-filesystem.md [hunk 4]
```diff
@@ -2,7 +2,105 @@
...
+      resolve: `gatsby-plugin-typography`,
+      options: {
+        pathToConfigModule: `src/utils/typography`,
+      },
+    },
+  ],
+}
+```
+
+Save that and restart the gatsby development server. Then open up Graph_i_QL
```
Comment (amberleyromo) [inline docs/docs/sourcing-from-the-filesystem.md]: We're not doing a step by step tutorial here -- I think what we could do here is

1. show an example file that gets sourced
2. show how to include the plugin (already done above)
3. show an example GraphiQL query (without explaining GraphiQL).

When we use a resource like GraphiQL here, we can link directly to where it's explained in other sections. For example, "Not familiar with GraphiQL? Check out this section of the tutorial that introduces it."

File: docs/docs/sourcing-from-the-filesystem.md [hunk 5]
```diff
@@ -2,7 +2,105 @@
...
+Try adding a number of fields to your query, pressing <kbd>Ctrl + Enter</kbd>
+each time to re-run the query. You'll see something like this:
+
+![allfile-query](images/allfile-query.png)
+
+The result is an array of File "nodes" (node is a fancy name for an object in a
+"graph"). Each File object has the fields you queried for.
+
+
+## What's coming next?
```
Comment (amberleyromo) [inline docs/docs/sourcing-from-the-filesystem.md]: Same thing here -- this only makes sense in the context of the tutorial.

**Comment by amberleyromo** [comment]
Hi @caseychien! Interested in tackling these comments and closing this out?

**Comment by caseychien** [comment]
Hi @amberleyromo ! I am a bit busy with school, it would probably be best if someone else could finish cleaning up this doc

**Comment by baumant** [comment]
@amerleyromo / @caseychien I'm happy to jump in here

**Comment by amberleyromo** [comment]
@shannonbux per https://github.com/gatsbyjs/gatsby/pull/8777#issuecomment-431178281 I cleaned up this doc further and it's ready for review / any further tweaks you wish!

**Comment by shannonbux** [comment]
closes #8730

**Pull request merged**